experimental/traffic-portal/src/app/shared/navigation/navigation.service.ts

```diff
@@ -60,8 +61,12 @@ export class NavigationService {
 
 	private readonly horizontalNavs: Map<string, HeaderNavigation>;
 	private readonly verticalNavs: Map<string, HeaderNavigation>;
+	private readonly tpv1Url: string | null = "http:localhost:433";
 
-	constructor(private readonly auth: CurrentUserService, private readonly api: UserService) {
+	constructor(
+		private readonly auth: CurrentUserService,
+		private readonly api: UserService) {
+		this.tpv1Url = localStorage.getItem(LOCAL_TPV1_URL) ? localStorage.getItem(LOCAL_TPV1_URL) : this.tpv1Url;
```

Comment:
This needs to be `window.localStorage`.

Since tpv1Url doesn't need to be nullable I'd just call `...getItem(...) ?? "";` instead.

Finally, this code will error out when the server tries to execute this component (because window is undefined). To solve this you'll need to inject `PLATFORM_ID` like you did in `app.component.ts` and check if we are `isPlatformBrowser` and otherwise set to blank string.